python/graphstorm/gconstruct/utils.py

```diff
@@ -36,6 +36,37 @@
 SHARED_MEM_OBJECT_THRESHOLD = 1.9 * 1024 * 1024 * 1024 # must < 2GB
 SHARED_MEMORY_CROSS_PROCESS_STORAGE = "shared_memory"
 PICKLE_CROSS_PROCESS_STORAGE = "pickle"
+EXT_MEMORY_STORAGE = "ext_memory"
+
+def _to_ext_memory(name, data, path):
+    if isinstance(data, np.ndarray):
+        assert name is not None
+        path = os.path.join(path, f"{name}.npy")
+        ext_mem_arr = convert_to_ext_mem_numpy(path, data)
+        # We need to pass the array to another process. We don't want it
+        # to reference to data in the file.
+        ext_mem_arr.cleanup()
```

Comment:
Should we just return the path to the file instead of the object?

Reply:
ext_mem_arr is a ExtNumpyWrapper. after cleaning up, the main thing it references is the path of the file and related metadata (shape, dtype).